data/src/main/java/org/oppia/android/data/backends/gae/RemoteAuthNetworkInterceptor.kt

```diff
@@ -0,0 +1,48 @@
+package org.oppia.android.data.backends.gae
+
+import android.content.Context
+import okhttp3.Interceptor
+import okhttp3.Request
+import okhttp3.Response
+import org.oppia.android.app.utility.getVersionCode
+import org.oppia.android.app.utility.getVersionName
+import java.io.IOException
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/**
+ * Interceptor on top of Retrofit to modify outgoing requests for authenticating messages.
+ *
+ * The Interceptor adds header parameters to outgoing messages.
+ */
+@Singleton
+class RemoteAuthNetworkInterceptor @Inject constructor() : Interceptor {
+
+  @Inject
+  lateinit var context: Context
+
+  @JvmField
+  @field:[Inject NetworkApiKey]
+  var networkApiKey: String = ""
+
+  @Throws(IOException::class)
+  override fun intercept(chain: Interceptor.Chain): Response {
+    val request = addAuthHeaders(chain.request())
+    return chain.proceed(request)
+  }
+
+  /**
+   * This function accepts the outgoing request and adds header values to it before sending.
+   *
+   * @param request the request being sent over the network
+   * @return the exact same request as the parameter with added headers
+   */
+  fun addAuthHeaders(request: Request): Request {
```

Comment:
Can this be private? It seems to only be needed for the class implementation.

Reply:
Is there a way to declare this as test-accessible? The network interceptor tests use this to verify the correct headers are added.

Reply:
I made this an internal fun so that the associated test could call on it and verify that headers are added correctly. I think another way to test this and keep the function private would be calling `intercept()` on a chain and verifying the response's associated request matched the interceptor's changes, but this is a bit more indirect (and probably more difficult) and I'm not sure if it's an accurate representation of the request anymore after it goes through the request. Using an internal function seems to be the most beneficial, WDYT?

Reply:
It looks like this is resolved now.